code/modules/organs/organ.dm

```diff
@@ -172,7 +172,7 @@ var/list/organ_cache = list()
 	handle_organ_proc_special()
 
 	//Process infections
-	if(robotic >= ORGAN_ROBOT || (istype(owner) && (owner && owner.species && (owner.species.flags & IS_PLANT || (owner.species.flags & NO_INFECT)))))
+	if(robotic >= ORGAN_ROBOT || (istype(owner) && (owner.species && (owner.species.flags & IS_PLANT || (owner.species.flags & NO_INFECT)))))
```

Comment:
These are also bitflags being checked, such that `(owner.species.flags & IS_PLANT) || (owner.species.flags & NO_INFECT)` is equivalent to `(owner.species.flags & (IS_PLANT | NO_INFECT))` (Note the single | for logical OR)